server/src/event/event.gateway.ts

```diff
@@ -59,6 +60,28 @@ export class EventGateway {
     }
   }
 
+  @SubscribeMessage('requestFilteredEventIds')
+  async requestFilteredEventIds(
+    @UserAbility() ability: AppAbility,
+    @CallingUser() user: User,
+    @MessageBody() data: RequestFilteredEventsDto,
+  ) {
+    const evs = await this.eventService.getEventsByIdsForAbility(
+      ability,
+      data.filterId,
+    );
+
+    console.log(evs.length);
+    for (const ev of evs) {
+      if (ev.difficulty == data.difficulty[0]) {
+        // if (ev.difficulty == "EASY") {
```

Comment:
Probably not amazing to have this code or the logging below unless this is a work in progress.